binder/src/main/java/io/grpc/binder/BinderChannelBuilder.java

```diff
@@ -280,6 +280,64 @@ public BinderChannelBuilder preAuthorizeServers(boolean preAuthorize) {
     return this;
   }
 
+  /**
+   * Specifies how and when to authorize a server against this Channel's {@link SecurityPolicy}.
+   *
+   * <p>This method selects the original "legacy" authorization strategy, which is no longer
+   * preferred for two reasons: First, the legacy strategy considers the UID of the server *process*
+   * we connect to. This is problematic for services using the `android:isolatedProcess` attribute,
+   * which runs them under a different "ephemeral" UID. This UID lacks all the privileges of the
+   * hosting app -- any non-trivial SecurityPolicy would fail to authorize it. Second, the legacy
+   * authorization strategy performs SecurityPolicy checks later in the connection handshake, which
+   * means the calling UID must be rechecked on every subsequent RPC. For these reasons, prefer
+   * {@link #useV2AuthStrategy} instead.
+   *
+   * <p>The server does not know which authorization strategy a client is using. Both strategies
+   * work with all versions of the grpc-binder server.
+   *
+   * <p>The default authorization strategy is unspecified. Clients that require the legacy strategy
+   * should configure it explicitly using this method. Eventually support for the legacy strategy
+   * will be removed.
+   *
+   * @return this
+   */
+  public BinderChannelBuilder useLegacyAuthStrategy() {
```

Comment:
Would it make sense to mark this `@Deprecated` or [@Discouraged](https://developer.android.com/reference/androidx/annotation/Discouraged) (if Jetpack is available) to point new integrators to the new API?

Or maybe you prefer doing so once the implementation is complete.

Reply:
Yes, eventually it will be deprecated and removed. But not yet because the replacement v2 hasn't proved itself in production yet. Have a look at the "Migration Process" section of the design doc for the proposed stages.